apache2/modsecurity.c

```diff
@@ -213,6 +213,8 @@ int modsecurity_init(msc_engine *msce, apr_pool_t *mp) {
         //return HTTP_INTERNAL_SERVER_ERROR;
         return -1;
     }
+
+    filelog_lock = msce->wafjsonlog_lock;
```

Comment:
Why is the filelog_lock being initialized from the msce->wafjsonlog_lock? This seems a bit round about? We can't initialize the filelog_lock directly?

Reply:
we have bunch of locks and they are all created in the modsec engines, this is the only lock we want to expose as a global lock. To keep consistent with other lock, this lock is also created within modsec->wafjsonlog_lock, and just get the handler of that.  Otherwise we should do the same thing for all the locks. which are not needed to be exposed as global variable.

Reply:
@azhao155 the code is not very idiomatic. Just because we are already creating locks in the modsecurity engine does not give us enough reason to couple the creation of global lock along with the other locks its very confusing from a readability standpoint. 

We should initialize the lock using the global variable explicitly and the lock shouldn't have any references in the modsecurit engine since it is global.